app/assets/javascripts/validation.js

```diff
@@ -0,0 +1,144 @@
+// Config
+/* global $ */
+var defaultErrorHeading = 'There\'s been a problem'
+var defaultErrorDescription = 'Check the following'
+var defaultErrorMessage = 'There is an error'
+
+function clearValidation () {
+  $('.error-summary').remove()
+
+  $('.error').each(function () {
+    $(this).removeClass('error')
+  })
+
+  $('.error-message').each(function () {
+    $(this).remove()
+  })
+}
+
+function checkTextFields (errors) {
+  $(document).find('input[type="text"], textarea').each(function () {
+    var $fieldset = $(this).parents('fieldset')
```

Comment:
you use $(this) multiple times in this block, might be better to do `var $this = $(this)` once at the start